File: python/misc/fact_trail_zero/fact_trail_zero.py
def fact_trail_zero(n):
    factor = 5
    count = 0
    while(n/factor >= 1):
        count += int(n/factor)
        factor *= 5
    return count

if __name__ == '__main__':
    import sys
    for line in sys.stdin:
        print(fact_trail_zero(int(line.strip())))

File: cpp/misc/fact_trail_zero/fact_trail_zero.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;

int fact_trail_zero(int n) {
    int factor = 5;
    int count = 0;
    while(n/factor >= 1) {
        count += n / factor;
        factor *= 5;
    }
    return count;
}

int main() {
    int n;
    while(cin >> n) {
        cout << fact_trail_zero(n) << endl;
    }
}
